lib/compile.js

```diff
@@ -105,6 +106,26 @@ function compileLoad(loader, load, compileOpts, cache) {
     if (format == 'cjs')
       mappedLoad.source = mappedLoad.source.replace(hashBangRegEx, '');
     return Promise.resolve()
+    .then(function () {
```

Comment:
I think we should move this whole implementation to be an extension of the loader itself, applied in the fetch hook at https://github.com/systemjs/builder/blob/master/lib/builder.js#L195. This way, plugins will be able to respond to input sourceMaps during builds if they wish to as well.